app/utilities/schema.py

```diff
@@ -27,34 +29,35 @@
 }
 
 
-def get_schema_path_map_for_language(language_code):
-    schema_files = []
-
-    for schema_dir in DEFAULT_SCHEMA_DIRS:
-        schema_files.extend(glob(f"{schema_dir}/{language_code}/*.json"))
+def get_schema_path_map_for_language(
+    language_code: str, dirs: Optional[Tuple[str, ...]] = None
+) -> Mapping:
 
     return {
         Path(schema_file).with_suffix("").name: schema_file
-        for schema_file in schema_files
+        for schema_dir in dirs or DEFAULT_SCHEMA_DIRS
+        for schema_file in glob(f"{schema_dir}/{language_code}/*.json")
     }
 
 
-def get_schema_path_map():
-    language_map_codes = ["en", "cy", "ga", "eo"]
+@lru_cache(maxsize=None)
+def get_schema_path_map(dirs: Optional[Tuple[str, ...]] = None) -> Mapping:
+    language_map_codes = {"en", "cy", "ga", "eo"}
 
     return {
-        language_code: get_schema_path_map_for_language(language_code)
+        language_code: get_schema_path_map_for_language(
+            language_code, dirs or DEFAULT_SCHEMA_DIRS
+        )
```

Comment:
I think the schemas endpoint should use this method rather than `get_schema_path_map_for_language` so that it gets the benefit of using the cached list rather than walking the filesystem again. If we do that, I think it's worth folding the `get_schema_path_map_for_language` method into this one, something like:

```python
    language_codes = {"en", "cy", "ga", "eo"}

    schema_path_map = {}
    for language_code in language_codes:
        schema_path_map[language_code] = {
            Path(schema_file).with_suffix("").name: schema_file
            for schema_dir in dirs or DEFAULT_SCHEMA_DIRS
            for schema_file in glob(f"{schema_dir}/{language_code}/*.json")
        }

    return schema_path_map
```

Reply:
Done, this is now 1 method. I decided to make the whole thing a dict comprehension, let me know what you think in terms of readability, personally I think it is just as readable.

Reply:
The dict comprehension is fine, I think the language codes should be defined as a constant though.